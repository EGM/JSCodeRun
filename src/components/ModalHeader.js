
import React from 'react';
import { 
    StyleSheet, 
    Text,
    View,
    TouchableOpacity,
    Platform
} from 'react-native';
import fonts from '../themes/fonts';

const styles = StyleSheet.create({
    modalHeader: {
        flexDirection: 'row',
        paddingHorizontal: 10,
        paddingTop: Platform.OS === 'ios' ? 22 : 2,
        paddingBottom: 10,
        backgroundColor: '#2b2b2b'
    },
    modalItem: {
        flex: 1,
        justifyContent: 'center'
    },
    modalTitle: {
        flex: 2,
        alignItems: 'center',
    },
    modalTitleText: {
        fontSize: 24,
        fontFamily: fonts.fontFamily,
        color: 'white',
    },
    modalClose: {
        flex: 1,
        alignItems: 'flex-end'
    },
    modalCloseText: {
        fontSize: 30,
        color: 'white'
    }
});

export default ModalHeader = props =>
    <View style={styles.modalHeader}>
        <View style={styles.modalItem}></View>
        <View style={[styles.modalItem, styles.modalTitle]}>
            <Text style={styles.modalTitleText}>{props.title}</Text>
        </View>
        <TouchableOpacity style={[styles.modalItem, styles.modalClose]} onPress={props.onClose}>
            <Text style={styles.modalCloseText}>âœ•</Text>
        </TouchableOpacity>
    </View>