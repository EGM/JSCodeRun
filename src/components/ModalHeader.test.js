import React, { Component } from 'react';
import renderer from 'react-test-renderer';
import ModalHeader from './ModalHeader';

describe('ModalHeader', () => {
  it('should render the ModalHeader correctly', () => {
    const props = {
      title: 'test',
      onClose: () => jest.fn()
    };
    const rendered = renderer.create(<ModalHeader {...props} />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
