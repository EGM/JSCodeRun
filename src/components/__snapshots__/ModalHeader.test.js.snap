// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalHeader should render the ModalHeader correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#2b2b2b",
      "flexDirection": "row",
      "paddingBottom": 10,
      "paddingHorizontal": 10,
      "paddingTop": 22,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "justifyContent": "center",
        },
        Object {
          "alignItems": "center",
          "flex": 2,
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontFamily": "HelveticaNeue",
          "fontSize": 24,
        }
      }
    >
      test
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "flex-end",
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontSize": 30,
        }
      }
    >
      âœ•
    </Text>
  </View>
</View>
`;
